forui/lib/src/foundation/tappable.dart

```diff
@@ -1,13 +1,36 @@
 import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
 import 'package:meta/meta.dart';
 
-// TODO: Remove redundant comment when flutter fixes its lint issue.
+@internal
+extension Touch on Never {
+  /// The platforms that uses touch as the primary input. It isn't 100% accurate as there are hybrid devices that uses
+  /// both touch and keyboard/mouse input, i.e. Windows Surface laptops.
+  static const platforms = {TargetPlatform.android, TargetPlatform.iOS, TargetPlatform.fuchsia};
+
+  static bool? _primary;
+
+  /// True if the current platform uses touch as the primary input.
+  static bool get primary => _primary ?? platforms.contains(defaultTargetPlatform);
+
+  @visibleForTesting
+  static set primary(bool? value) {
+    if (!kDebugMode) {
+      throw UnsupportedError('Setting Touch.primary is only available in debug mode.');
+    }
+
+    _primary = value;
+  }
+}
+
+/// The [FTappableState]'s current state.
 ///
+/// Short pressed is fired 200 ms after a tap is first detected. This is faster than long press's [kLongPressTimeout].
 @internal
-typedef FTappableState = ({bool focused, bool hovered, bool longPressed});
+typedef FTappableState = ({bool focused, bool hovered, bool shortPressed});
```

Comment:
Ehhh what's `shortPressed` for? Why can't we just stick with a `longPressed`?

Reply:
I renamed it from longPress because it's technically shorter than longPress, 200ms instead of 500ms. The reason being that long press takes too long and feels unresponsive. I'm open to names though.